pkg/cloud/azure/actuators/machine/conditions.go

```diff
@@ -0,0 +1,104 @@
+package machine
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/apis/azureprovider/v1alpha1"
+)
+
+const (
+	// MachineCreationSucceeded indicates success for machine creation
+	MachineCreationSucceeded = "MachineCreationSucceeded"
+
+	// MachineCreationFailed indicates that machine creation failed
+	MachineCreationFailed = "MachineCreationFailed"
+)
+
+// updateConditionCheck tests whether a condition should be updated from the
+// old condition to the new condition. Returns true if the condition should
+// be updated.
+type updateConditionCheck func(oldReason, oldMessage, newReason, newMessage string) bool
+
+// updateConditionIfReasonOrMessageChange returns true if there is a change
+// in the reason or the message of the condition.
+func updateConditionIfReasonOrMessageChange(oldReason, oldMessage, newReason, newMessage string) bool {
+	return oldReason != newReason ||
+		oldMessage != newMessage
+}
+
+func shouldUpdateCondition(
+	oldStatus corev1.ConditionStatus, oldReason, oldMessage string,
+	newStatus corev1.ConditionStatus, newReason, newMessage string,
+	updateConditionCheck updateConditionCheck,
+) bool {
+	if oldStatus != newStatus {
```

Comment:
why do we need this?
can we just have a function which check the three of them? https://github.com/openshift/cluster-api-provider-gcp/blob/master/pkg/cloud/gcp/actuators/machine/conditions.go#L15

Reply:
Since we might update the condition on multiple places/phases in the future with different condition.

Reply:
can we include the abstraction if ever at all the case you describe is needed?

Reply:
I will keep this function here since it names the expression `(oldStatus != newStatus) || (oldReason != newReason || oldMessage != newMessage)`. Without the name it's just "hard to read expression".